### PR TITLE
build: use bazel for snapshot builds

### PR DESCRIPTION
**Pull request by kormide**
@gregmagolan @josephperrott 

Here are the snapshot repo diffs caused by flipping to the bazel build.
| Snapshot repo  | Diff |
| ------------- | ------------- |
| @angular/cli  | [diff](https://github.com/angular/cli-builds/compare/master...snapshot-test)  |
| @angular/pwa  | [diff](https://github.com/angular/angular-pwa-builds/compare/master...snapshot-test)  |
| @angular-devkit/architect | [diff](https://github.com/angular/angular-devkit-architect-builds/compare/master...snapshot-test)  |
| @angular-devkit/architect-cli | [diff](https://github.com/angular/angular-devkit-architect-cli-builds/compare/master...snapshot-test)  |
| @angular-devkit/build-angular | [diff](https://github.com/angular/angular-devkit-build-angular-builds/compare/master...snapshot-test)  |
| @angular-devkit/build-webpack | [diff](https://github.com/angular/angular-devkit-build-webpack-builds/compare/master...snapshot-test)  |
| @angular-devkit/core | [diff](https://github.com/angular/angular-devkit-core-builds/compare/master...snapshot-test)  |
| @angular-devkit/schematics | [diff](https://github.com/angular/angular-devkit-schematics-builds/compare/master...snapshot-test)  |
| @angular-devkit/schematics-cli | [diff](https://github.com/angular/angular-devkit-schematics-cli-builds/compare/master...snapshot-test)  |
| @ngtools/webpack | [diff](https://github.com/angular/ngtools-webpack-builds/compare/master...snapshot-test)  |
| @schematics/angular | [diff](https://github.com/angular/schematics-angular-builds/compare/master...snapshot-test)  |

Note that some changes in these diffs may be caused by new things pushed to `master` of the "build" repos.

[CI runs](https://app.circleci.com/pipelines/github/angular/angular-cli?branch=snapshot-test&filter=all) for the `snapshot-test` branch set up to test the ci. The build artifacts are published to the build repos under the same `snapshot-test` branch name.

### CODE REVIEW

**Review by gregmagolan — APPROVED** [review]

File: lib/packages.ts
Comment (kormide) [inline lib/packages.ts:163]: In the snapshot script, there's a `process.chdir()` outside of the git repo, which causes this calculation of the hash to fail. This just makes sure it gets run inside of the git repo. This didn't occur previously because we loaded the legacy build script cached the sha value before that chdir occurred.

File: .circleci/config.yml
Comment (kormide) [inline .circleci/config.yml:273]: The tarring that bazel required python.

File: tools/snapshot_repo_filter.bzl
Comment (kormide) [inline tools/snapshot_repo_filter.bzl:11]: Bug in the github dependency format.

**Review by josephperrott — APPROVED** [review]
LGTM

**Comment by kormide** [comment]
I'd like to run a few more tests before we mark this as ready to merge.

**Comment by kormide** [comment]
I ran into an issue trying to manually install the snapshots. The bazel build stamps snapshot versions of dependencies like:
```
github:angular/angular-devkit-architect-builds#bb8bc80581417898b10bdc23ef4d9ee4295da2a9
```
It's using the stamp variable `{BUILD_SCM_COMMIT_SHA}` provided by the `ng-dev` tool.

However, legacy build stamps using an abbreviated version of the SHA1, and the snapshot script tags the commit in the build repo using the abbreviated version, so running an npm install of the snapshots fails.

I tried changing the snapshot script to tag using the longer hash, however git complains that the name is too long.

Something I could do is to change the `dev-infra` repo to output another stamp with the abbreviated hash so that we can stamp that instead. Wdyt @josephperrott?

**Comment by alan-agius4** [comment]
Question why is the version `13.2.0-next` and not `14.0.0-next`?

Screenshot 2022-01-31 at 13 06 35

**Comment by kormide** [comment]
>     * why is the version `13.2.0-next` and not `14.0.0-next`?

I think that's just because `master` in the build repo has a more recent snapshot that the branch I'm working off of. That should resolve itself the next time a snapshot it pushed on `snapshot-test`.

> 
>     * can we also turn off source-map generation?
Unfortunately I don't think we can with `ts_library`. @josephperrott previously ok'ed this, and it's consistent with what the `angular` repo produces.

**Comment by alan-agius4** [comment]
> Unfortunately I don't think we can with ts_library. @josephperrott previously ok'ed this, and it's consistent with what the angular bazel build produces.

This is not a big of deal, although sourcemaps are a bit redundant here and kinda of extra mbs over the wire and disk in this case. Unlike for angular/angular were they are useful for app developers. 

But yeah, I don't think there is away to disable sourcemaps with `ts_library`.

**Comment by kormide** [comment]
I'm now able to install the snapshot build locally. I'm getting a failure on the e2e windows tests, which I've seen in other ci runs so it appears unrelated to my changes. Is that test suite known to be flaky? https://app.circleci.com/pipelines/github/angular/angular-cli/20377/workflows/dc026fcf-1783-4d3d-9ac9-9ec9854c1eb3/jobs/286136

**Comment by alan-agius4** [comment]
@kormide, looks like our Node.Js v12 CI is red.

**Comment by kormide** [comment]
> @kormide, looks like our Node.Js v12 CI is red.
> 
> Will investigate tomorrow, however it’s nothing related to this change.

In that case, I think this is ready to go, unless you can think of anything else I should test first.

**Comment by alan-agius4** [comment]
Do you want to wait for @gregmagolan’s review?

**Comment by kormide** [comment]
> Do you want to wait for @gregmagolan’s review?

Your and @josephperrott's review is sufficient. @gregmagolan reviewed previously and is in the loop, but hes out today.

**Review by alan-agius4 — APPROVED** [review]
LGTM

**Comment by dgp1130** [comment]
There was a trivial conflict with `package.json` / `yarn.lock` in `13.2.x`. So I merged this to `master` and did a manual cherry pick to `13.2.x` in https://github.com/angular/angular-cli/commit/491e2103bd52bc20c886f44d00fdad02c2124b55.

**Pull request merged**
